src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1429,48 +1409,18 @@ public void RegisterInputPorts()
                 for (int i = inPorts.Count - 1; i >= count; i--)
                     inPorts.RemoveAt(i);
             }
-
-            //Configure Snap Edges
-            ConfigureSnapEdges(inPorts);
-            areInputPortsRegistered = true;
-
-            RaisesModificationEvents = true;
-            OnNodeModified();
         }
 
         /// <summary>
         ///     Reads outputs list and adds ports for each output
         /// </summary>
         [Obsolete("RegisterOutputPorts is deprecated, please use the OutPortNamesAttribute, OutPortDescriptionsAttribute, and OutPortTypesAttribute instead.")]
-        public void RegisterOutputPorts()
+        public void RegisterOutputPorts(IEnumerable<PortData> portDatas)
```

Comment:
API Change: `NodeModel.RegisterOutputPorts()` is now `NodeModel.RegisterOutputPorts(IEnumerable<PortData> portDatas)`